lxd/seccomp/seccomp.go

```diff
@@ -926,16 +901,28 @@ func (s *Server) HandleMknodSyscall(c Instance, siov *Iovec) int {
 			"seccomp_notify_flags": siov.req.flags,
 		})
 
-	siov.resp.error = C.device_allowed(C.dev_t(siov.req.data.args[2]), C.mode_t(siov.req.data.args[1]))
-	if siov.resp.error != 0 {
+	if C.device_allowed(C.dev_t(siov.req.data.args[2]), C.mode_t(siov.req.data.args[1])) < 0 {
 		logger.Debugf("Device not allowed")
+		if s.s.OS.SeccompListenerContinue {
+			logger.Debugf("Continuing mknod syscall")
+			C.seccomp_notify_update_response(siov.resp, 0, C.uint32_t(seccompUserNotifFlagContinue))
+			return 0
+		}
```

Comment:
Can you reshuffle things a little so we don't en up logging two messages per syscall in the continue case?

Reply:
And do the same to all the other cases below?

Reply:
completely reworked